src/libmongoc/tests/test-mongoc-dns.c

```diff
@@ -522,22 +599,158 @@ test_small_initial_buffer (void *unused)
    _mongoc_host_list_destroy_all (rr_data.hosts);
 }
 
+bool
+_mock_resolver (const char *service,
+                mongoc_rr_type_t rr_type,
+                mongoc_rr_data_t *rr_data,
+                size_t initial_buffer_size,
+                bson_error_t *error)
+{
+   rr_data->count = 2;
+   rr_data->hosts = MAKE_HOSTS ("localhost.test.build.10gen.cc:27017",
+                                "localhost.test.build.10gen.cc:27018");
+   rr_data->min_ttl = 1;
+   rr_data->txt_record_opts = NULL;
+   return true;
+}
+
+static void
+_prose_loadbalanced_ping (mongoc_client_t *client)
+{
+   bson_error_t error;
+   bson_t *cmd = BCON_NEW ("ping", BCON_INT32 (1));
+
+   if (!mongoc_client_command_simple (
+          client, "admin", cmd, NULL, NULL, &error)) {
+      test_error ("ping failed: %s", error.message);
+   }
+
+   bson_destroy (cmd);
+}
+
+/*
+   Implements the following prose test described in the SRV polling test README:
+   Test that SRV polling is not done for load balalanced clusters. Connect to
+   mongodb+srv://test3.test.build.10gen.cc/?loadBalanced=true, mock the addition
+   of the following DNS record, wait until 2*rescanSRVIntervalMS, and assert
+   that the final topology description only contains one server
+   (localhost.test.build.10gen.cc. at port 27017).
+   _mongodb._tcp.test3.test.build.10gen.cc. 86400 IN SRV 27018
+   localhost.test.build.10gen.cc.
+*/
+static void
+_prose_loadbalanced_run (bool pooled)
+{
+   mongoc_client_pool_t *pool;
+   mongoc_uri_t *uri;
+   mongoc_client_t *client;
+   mongoc_host_list_t *expected_hosts;
+   mongoc_topology_t *topology;
+#define RESCAN_INTERVAL_MS 500
+#ifdef MONGOC_ENABLE_SSL
+   mongoc_ssl_opt_t ssl_opts = *test_framework_get_ssl_opts ();
+
+   ssl_opts.allow_invalid_hostname = true;
+#endif
+
+
+   uri = mongoc_uri_new ("mongodb+srv://test3.test.build.10gen.cc");
+   mongoc_uri_set_option_as_bool (uri, MONGOC_URI_LOADBALANCED, true);
+   mongoc_uri_set_option_as_int32 (
+      uri, MONGOC_URI_HEARTBEATFREQUENCYMS, RESCAN_INTERVAL_MS);
```

Comment:
The prose test doesn't talk about tweaking heartbeatFrequencyMS. What's the rationale for doing this?